specification/security/resource-manager/Microsoft.Security/stable/2024-08-01/customRecommedations.json

```diff
@@ -0,0 +1,448 @@
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "describes Custom Recommendation properties.",
+          "$ref": "#/definitions/CustomRecommendationProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "CustomRecommendationProperties": {
+      "type": "object",
+      "description": "describes the Custom Recommendation properties",
+      "properties": {
+        "query": {
+          "description": "KQL query representing the Recommendation results required.",
+          "type": "string"
+        },
+        "cloudProviders": {
+          "description": "List of all standard supported clouds.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RecommendationSupportedCloud"
+          }
+        },
+        "severity": {
+          "description": "The severity to relate to the assessments generated by this Recommendation.",
+          "type": "string",
+          "enum": [
+            "High",
+            "Medium",
+            "Low"
+          ],
+          "x-ms-enum": {
+            "name": "severityEnum",
+            "modelAsString": true
+          },
+          "x-ms-client-default": "Low"
+        },
+        "displayName": {
+          "description": "The display name of the assessments generated by this Recommendation.",
+          "type": "string"
+        },
+        "description": {
+          "description": "The description to relate to the assessments generated by this Recommendation.",
+          "type": "string"
+        },
+        "remediationDescription": {
+          "description": "The remediation description to relate to the assessments generated by this Recommendation.",
+          "type": "string"
+        },
+        "assessmentKey": {
+          "description": "The assessment metadata key used when an assessment is generated for this Recommendation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "securityIssue": {
+          "description": "The severity to relate to the assessments generated by this Recommendation.",
+          "type": "string",
+          "enum": [
+            "Vulnerability",
+            "ExcessivePermissions",
+            "AnonymousAccess",
+            "NetworkExposure",
+            "TrafficEncryption",
+            "BestPractices"
+          ],
+          "x-ms-enum": {
+            "name": "securityIssue",
+            "modelAsString": true
+          },
+          "x-ms-client-default": "BestPractices"
+        },
+        "systemData": {
+          "readOnly": true,
+          "type": "object",
+          "description": "Azure Resource Manager metadata containing createdBy and modifiedBy information.",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
```

Comment:
>v2 [](http://example.com/codeflow?start=67&length=2)

Should reference the newest common types (v6).

Reply:
done